services/controller/src/plz/controller/main.py

```diff
@@ -250,6 +263,32 @@ def delete_execution(execution_id):
     return response
 
 
+@app.route(f'/executions/<user>/<project>/history', methods=['GET'])
+def history_entrypoint(user, project):
+    execution_ids = db_storage.retrieve_execution_ids_for_user_and_project(
+        user, project)
+
+    @stream_with_context
+    def act():
+        yield '{\n'
+        first = True
+        for execution_id in execution_ids:
+            try:
+                metadata = get_metadata(execution_id)
+            except ExecutionIdNotFound as e:
+                log.info(e)
```

Comment:
Surely `log.warning` at least?

You might want to use `log.exception('Missing execution ID.')` for this one, because it knows how to format an exception. You _don't_ need to pass any arguments apart from an message; it picks up the exception value from the context.

Reply:
Well, not really. The executions we have right don't have any metadata, that's not bad. Maybe what we can do is to return an empty map to the user, so that at least these executions didn't just disappear?

Reply:
Oh, but the executions we have so far won't be in the db storage, so it's not a problem in the end. I've hit the problem at some point (probably while already had the db implemented but not the metadata). I'm just throwing an exception now so it fails loudly.

Reply:
Fair enough.